### PR TITLE
consensus: update state from store before use in reactor

### PR DESCRIPTION
**Pull request by williambanfield**
Closes: #8575

This PR aims to fix the `LastCommitRound can only be negative for initial height 0` issue we see in the e2e tests by initializing the `state` object before starting the receive routines in the consensus reactor. This is somewhat inelegant, but it should fix the issue.

### CODE REVIEW

File: internal/consensus/reactor.go [hunk 1]
```diff
@@ -216,6 +216,10 @@ func (r *Reactor) OnStart(ctx context.Context) error {
 		if err := r.state.Start(ctx); err != nil {
 			return err
 		}
+	} else {
+		if err := r.state.updateStateFromStore(); err != nil {
+			return err
+		}
 	}
```
Comment (creachadair) [inline internal/consensus/reactor.go:223]:
```suggestion
	} else if err := r.state.updateStateFromStore(); err != nil {
		return err
	}
```

**Review by creachadair — APPROVED** [review]
This seems plausible to me.

**Comment by cmwaters** [comment]
Have you run this in the e2e framework to confirm the error no longer appears @williambanfield?

**Comment by williambanfield** [comment]
> Have you run this in the e2e framework to confirm the error no longer appears @williambanfield?

@cmwaters good question. Had run locally before and saw the error before this fix and no longer see it now. 

Ran manually also here and no longer see:
https://github.com/tendermint/tendermint/runs/6508583163?check_suite_focus=true

**Pull request merged**